### DISCUSSION TITLE
overlay spinner that doesn't hide the page content

### DISCUSSION THREAD

**Discussion by jeevan-sg** [twilio-labs/paste#2427] [General] [answered]
Hi Paste,

Is there an overlay spinner available? 
Currently spinner hides the content until data available, is there a way to implement overlay spinner that doesn't hide the page content instead displays as an overlay? Thanks

https://paste.twilio.design/components/spinner

**Comment by nkrantz** [comment]
Hi @jeevan-sg, could you provide some more context on what you're trying to achieve? Do you have any designs you're working off of that you could share screenshots or a link to? Wondering what the exact use case is for needing a spinner overlay.

**Comment by jeevan-sg** [comment, marked as answer]
Hi Nora,

I am referring to this image where content is visible in the background while spinner is on front. Can you advise how to implement this behavior using paste spinner? Thanks

<img width="554" alt="overlay-loader" src="https://user-images.githubusercontent.com/45638853/170121659-6cfd1141-171b-42e9-a89f-236e014e0c35.png">

**Reply by TheSisb** [reply]
Hi @jeevan-sg,

We don't have a generic overlay component in Paste, but you can see how we do it for our Modal component and create your own version: https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/modal/src/Modal.tsx#L16-L38

**Reply by nkrantz** [reply]
@jeevan-sg Just want to make sure you're aware of the existing components and patterns in Paste for loading pages. We'd recommend using the [Skeleton Loader](https://paste.twilio.design/components/skeleton-loader) for now, unless there's a reason the spinner overlay would work better for your use case.

**Reply by FirstWhack** [reply]
+1 on this, skeleton loader does not make sense for a table heading that changes for example, only the content itself.
